### PR TITLE
Add override for default region.

### PR DESCRIPTION
**Pull request by danmancuso**
Resolves Issue #27 

This change allows overriding of the default region by either adding an environment variable or by a setting in the secretsmanager.properties file. Some small refactoring of where we set the PrivateLink DNS endpoint configuration was done in the process. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by simonmarty — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -92,3 +92,20 @@ The secret being used should be in the JSON format we use for our rotation lambd
 	...
 }
 ```
+### Overriding Default Regions
+
+The default region provider chain is used if no adjustments are made. In order to override this default, use one of the following options listed below.
```
Comment (willtong1234) [inline README.md:97]: These provide a way to override the default region and I do like this change for other use cases (so don't revert it), but we also want a way for customers to configure a backup region in order to provide high availability using replicated secrets. There are multiple ways to accomplish this. Some potential options:

* Allow for a "backupRegion" to be specified via configuration here in the JDBC driver. If present, we can instantiate a second SecretCache with this region. Then in the driver logic, [if we can't retrieve the secret in the primary region](https://github.com/aws/aws-secretsmanager-jdbc/blob/master/src/main/java/com/amazonaws/secretsmanager/sql/AWSSecretsManagerDriver.java#L343) cache, we can fallback to using the backup SecretCache. There are a couple places where the secret is retrieved from the cache so we would want to split out this logic into a helper function.
* We can implement the fallback logic in the SecretCache itself. The cache could take a backup region in via [configuration](https://github.com/aws/aws-secretsmanager-caching-java/blob/master/src/main/java/com/amazonaws/secretsmanager/caching/SecretCacheConfiguration.java) and have two different AWSecretsManager clients. If the primary region fails for whatever reason, we can fallback to the backup region.

In both approaches, if a Secret ARN is provided, we probably want to do some automatic region substitution as well.

File: src/main/java/com/amazonaws/secretsmanager/util/JDBCSecretCacheBuilderProvider.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        //Retrieve data from information sources.
+        String vpcEndpointUrl = configFile.getStringPropertyWithDefault(AWSSecretsManagerDriver.PROPERTY_PREFIX+"."+PROPERTY_VPC_ENDPOINT_URL, null);
+        String vpcEndpointRegion = configFile.getStringPropertyWithDefault(AWSSecretsManagerDriver.PROPERTY_PREFIX+"."+PROPERTY_VPC_ENDPOINT_REGION, null);
+        String envRegion = System.getenv(REGION_ENVIRONMENT_VARIABLE);
+        String configRegion = configFile.getStringPropertyWithDefault(AWSSecretsManagerDriver.PROPERTY_PREFIX+"."+PROPERTY_REGION, null);
+
+
+        //Apply settings to our builder configuration.
+        if (vpcEndpointUrl != null && !vpcEndpointUrl.isEmpty() && vpcEndpointRegion != null && !vpcEndpointRegion.isEmpty()) {
```
Comment (willtong1234) [inline src/main/java/com/amazonaws/secretsmanager/util/JDBCSecretCacheBuilderProvider.java:66]: Nitpick, but we can use !isNullOrEmpty in com.amazonaws.util.StringUtils to reduce the number of checks in these if conditions.

Comment (danmancuso) [reply src/main/java/com/amazonaws/secretsmanager/util/JDBCSecretCacheBuilderProvider.java]: Updated!

**Review by willtong1234 — APPROVED** [review]
I'm approving the file change overall, but it would be great if you could improve the commit message. Previously those who used the constructor that took in a SecretCache would get the cache overwritten if they have endpoint/region properties specified. This is no longer true. I think it's an improvement, but since it is a backward incompatible change we should at least call that out in the commit message for transparency.

File: README.md (hunk 1 above)
Comment (lnjohns) [reply README.md:97]: This comment is spot on with our need. During DR testing, we would like to be able to configure failover secret/region config to be used.

**Pull request merged**